test/core/tsi/crl_ssl_transport_security_test.cc

```diff
@@ -54,103 +51,90 @@ const char* kSslTsiTestCrlSupportedCrlDirMissingIntermediate =
+    absl::StrCat(kSslTsiTestCrlSupportedCredentialsDir, "valid.key");
+const std::string kValidCertPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCredentialsDir, "valid.pem");
+const std::string kRevokedIntermediateKeyPath = absl::StrCat(
+    kSslTsiTestCrlSupportedCredentialsDir, "leaf_signed_by_intermediate.key");
+const std::string kRevokedIntermediateCertPath = absl::StrCat(
+    kSslTsiTestCrlSupportedCredentialsDir, "leaf_and_intermediate_chain.pem");
+const std::string kRootCrlPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCrlDir, "current.crl");
+const std::string kIntermediateCrlPath =
+    absl::StrCat(kSslTsiTestCrlSupportedCrlDir, "intermediate.crl");
+
+char* LoadFile(absl::string_view file_path) {
+  grpc_slice slice;
+  GPR_ASSERT(grpc_load_file(file_path.data(), 1, &slice) == absl::OkStatus());
+  char* data = grpc_slice_to_c_string(slice);
+  grpc_slice_unref(slice);
+  return data;
+}
 
 class CrlSslTransportSecurityTest
     : public testing::TestWithParam<tsi_tls_version> {
  protected:
   // A tsi_test_fixture implementation.
   class SslTsiTestFixture {
    public:
-    // When use_faulty_crl_directory is set, the crl_directory of the
-    // client is set to a non-existant path.
-    static SslTsiTestFixture* Create(bool use_revoked_server_cert,
-                                     bool use_revoked_client_cert,
-                                     bool use_faulty_crl_directory) {
-      return new SslTsiTestFixture(
-          use_revoked_server_cert, use_revoked_client_cert,
-          use_faulty_crl_directory, false, false, false);
-    }
-
-    static SslTsiTestFixture* CreateWithIntermediate(
-        bool use_revoked_intermediate, bool use_missing_intermediate_crl,
-        bool use_missing_root_crl) {
-      return new SslTsiTestFixture(
-          false, false, false, use_revoked_intermediate,
-          use_missing_intermediate_crl, use_missing_root_crl);
-    }
-
-    void Run() {
-      tsi_test_do_handshake(&base_);
-      tsi_test_fixture_destroy(&base_);
-    }
-
-   private:
-    SslTsiTestFixture(bool use_revoked_server_cert,
-                      bool use_revoked_client_cert,
-                      bool use_faulty_crl_directory,
-                      bool use_revoked_intermediate,
-                      bool use_missing_intermediate_crl,
-                      bool use_missing_root_crl)
-        : use_revoked_server_cert_(use_revoked_server_cert),
-          use_revoked_client_cert_(use_revoked_client_cert),
-          use_faulty_crl_directory_(use_faulty_crl_directory),
-          use_revoked_intermediate_(use_revoked_intermediate),
-          use_missing_intermediate_crl_(use_missing_intermediate_crl),
-          use_missing_root_crl_(use_missing_root_crl) {
+    SslTsiTestFixture(
+        const std::string& server_key_path, const std::string& server_cert_path,
+        const std::string& client_key_path, const std::string& client_cert_path,
+        const char* crl_directory,
```

Comment:
nit: Can we use a `const std::string&` or `absl::string_view` instead of a `const char*`?